nomad/vault.go

```diff
@@ -578,6 +594,37 @@ func (v *vaultClient) getWrappingFn() func(operation, path string) string {
 	}
 }
 
+const vaultDateLayout = time.RFC3339Nano
+
+func parseTokenData(s map[string]interface{}) (tokenData, error) {
+	// Read and parse the fields
+	var data tokenData
+	if err := mapstructure.WeakDecode(s, &data); err != nil {
+		return data, fmt.Errorf("failed to parse Vault token's data block: %v", err)
+	}
+
+	root := false
+	for _, p := range data.Policies {
+		if p == "root" {
+			root = true
+			break
+		}
+	}
+
+	// Store the token data
+	data.Root = root
+
+	if data.ExpireTimeString != "" {
+		if d, err := time.Parse(vaultDateLayout, data.ExpireTimeString); err == nil {
+			data.ExpireTime = d
```

Comment:
Can you log if this fails and fall back to using the TTL

Reply:
I abondand relying on `expire_time` field, so this no longer relevant.